### PR TITLE
Add almalinux support

### PR DESCRIPTION
**Pull request by eabdullin1**
Hello, add support for AlmaLinux for dotnet

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=62184) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=62184) before we can accept your contribution.<br/><br/>:x: eabdullin1 [sign now](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=62184)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/runtime?pullRequest=62184) it.</sub>

File: eng/native/init-distro-rid.sh [hunk 1]
```diff
@@ -41,7 +41,7 @@ initNonPortableDistroRid()
             # We have forced __PortableBuild=0. This is because -portablebuld
             # has been passed as false.
             if (( isPortable == 0 )); then
-                if [[ "${ID}" == "rhel" || "${ID}" == "rocky" ]]; then
+                if [[ "${ID}" == "rhel" || "${ID}" == "rocky" || "${ID}" == "almalinux" ]]; then
                     # remove the last version digit
                     VERSION_ID="${VERSION_ID%.*}"
```
Comment (am11) [inline eng/native/init-distro-rid.sh:46]: Instead of accounting for each derivative distro separately, how about:
```suggestion
                # for RHEL and its derivatives, remove the last version digit (if present)
                if [[ "${ID}" == "rhel" || "${ID_LIKE}" =~ .*rhel.* ]]; then
                    VERSION_ID="${VERSION_ID%.*}"
```

**Comment by safern** [comment]
Adjusting the area as I think this belongs to the host more than libraries infra.

Infra changes look good, they match what was done in: https://github.com/dotnet/runtime/pull/59178

@eerhardt @vitek-karas @wfurt could you please help review?

File: eng/native/init-distro-rid.sh [hunk 2]
```diff
@@ -41,8 +41,8 @@ initNonPortableDistroRid()
             # We have forced __PortableBuild=0. This is because -portablebuld
             # has been passed as false.
             if (( isPortable == 0 )); then
-                if [[ "${ID}" == "rhel" || "${ID}" == "rocky" ]]; then
-                    # remove the last version digit
+                # for RHEL and its derivatives, remove the last version digit (if present)
+                if [[ "${ID}" == "rhel" || "${ID_LIKE}" =~ .*rhel.* ]]; then
```
Comment (eerhardt) [inline eng/native/init-distro-rid.sh:45]: does this also affect centos and fedora?

Comment (am11) [reply eng/native/init-distro-rid.sh:45]: The condition matches centos but not fedora.
After that value on centos is not affected by the line below, so it is only for distros which we have version / release policy like RHEL and are derivatives of that.

Comment (am11) [reply eng/native/init-distro-rid.sh:45]: Also, these PRs are not really needed, unless the distro really wants to create a "non-portable" release of .NET and provide some distro-specific packages.

https://dotnet.microsoft.com/download/dotnet/thank-you/sdk-6.0.100-linux-x64-binaries works for any glibc-based linux distro..

`non-portable` is only used by source builds for a long time now, elsewhere we all use portable binaries, so these PRs are less meaningful.

**Comment by eabdullin1** [comment]
Are there any updates?

**Comment by vitek-karas** [comment]
I know very little about this part of the Linux PAL in the host. @wfurt might have a better idea.
The code looks good, but I can't comment if it will work on the actual OS.

**Comment by eabdullin1** [comment]
@wfurt Hello. Some updates?

**Comment by andrewlukoshko** [comment]
Are there any other dotnet projects repos where we should do the same?

**Comment by am11** [comment]
Distro specific RID (for non-portable build; i.e. `./build.sh -portable=false`) in practice is only needed if a distro wants to provide and maintain distro nuget packages _and_ that specific package is not usable for other distros.

This is a very rare case when a distro needs that kind of specificity. It was used by default in .NET Core v1.x. Starting from v2, official builds moved to portable build plan. Therefore, today in the wild, the commonly used installations are the generic linux or linux-musl ones for the supported architectures.

See the download page for binary installers: https://dot.net. Those installers can be built from dotnet/source-build (just don't pass `portable=false` legacy option).

In other words:
> Add almalinux support

artifacts listed in table under https://github.com/dotnet/installer#installers-and-binaries should be functional without these changes.

**Comment by agocke** [comment]
Yeah, I'm not inclined to take this unless there's specific support needed for almalinux. Namely, what does the new RID provide that linux-x64 does not?

**Comment by danmoseley** [comment]
@agocke should we close this then?

**Comment by trylek** [comment]
Closing for now as there's been no traffic on this PR for almost three months. Please feel free to reopen to follow up on this.
